packages/celotool/src/e2e-tests/utils.ts

```diff
@@ -385,109 +125,155 @@ export async function migrateContracts(
+  spawnWithLog(bootnodeBinaryPath, bootnodeArgs, bootnodeLog, verbose)
+  return getEnodeAddress(privateKeyToPublicKey(bootnodePrivateKey), '127.0.0.1', 30301)
 }
 
-/**
- * @returns Promise<number> the geth pid number
- */
-export async function initAndStartGeth(gethBinaryPath: string, instance: GethInstanceConfig) {
-  const datadir = getDatadir(instance)
-  console.info(`geth:${instance.name}: init datadir ${datadir}`)
-  await init(gethBinaryPath, datadir, GENESIS_PATH)
-  if (instance.privateKey) {
-    await importPrivateKey(gethBinaryPath, instance)
-  }
-  if (instance.peers) {
-    await addStaticPeers(datadir, instance.peers)
+export async function assertRevert(promise: any, errorMessage: string = ''): Promise<void> {
+  try {
+    await promise
+    assert.fail('Expected revert not received')
+  } catch (error) {
+    const revertFound = error.message.search('revert') >= 0
+    if (errorMessage === '') {
+      assert(revertFound, `Expected "revert", got ${error} instead`)
+    } else {
+      assert(revertFound, errorMessage)
+    }
   }
-  return startGeth(gethBinaryPath, instance)
 }
 
-// Add validator 0 as a peer of each other validator.
-async function connectValidatorPeers(gethConfig: GethTestConfig) {
-  const admins = gethConfig.instances
-    .filter(({ wsport, rpcport, validating }) => validating && (wsport || rpcport))
-    .map(
-      ({ wsport, rpcport }) =>
-        new Admin(`${wsport ? 'ws' : 'http'}://localhost:${wsport || rpcport}`)
-    )
-  const enodes = await Promise.all(admins.map(async (admin) => (await admin.getNodeInfo()).enode))
-  await Promise.all(
-    admins.map(async (admin, i) => {
-      await Promise.all(
-        enodes.map(async (enode, j) => {
-          if (i === j) {
-            return
-          }
-          await admin.addPeer(enode)
-        })
-      )
-    })
-  )
+export function getHooks(gethConfig: GethRunConfig) {
+  return getContext(gethConfig).hooks
 }
 
-export function getContext(gethConfig: GethTestConfig) {
+export function getContext(gethConfig: GethRunConfig, verbose: boolean = verboseOutput) {
   const mnemonic =
     'jazz ripple brown cloth door bridge pen danger deer thumb cable prepare negative library vast'
   const validatorInstances = gethConfig.instances.filter((x: any) => x.validating)
+
   const numValidators = validatorInstances.length
+
   const validatorPrivateKeys = getPrivateKeysFor(AccountType.VALIDATOR, mnemonic, numValidators)
   const attestationKeys = getPrivateKeysFor(AccountType.ATTESTATION, mnemonic, numValidators)
-  const validators = getValidators(mnemonic, numValidators)
+  const validators = getValidatorsInformation(mnemonic, numValidators)
+
+  const proxyInstances = gethConfig.instances.filter((x: any) => x.isProxy)
+  const numProxies = proxyInstances.length
+
+  const proxyPrivateKeys = getPrivateKeysFor(AccountType.PROXY, mnemonic, numProxies)
+  const proxyEnodes = proxyPrivateKeys.map((x: any, i: number) => [
```

Comment:
Can't this just be `string` instead of `any`?
```suggestion
  const proxyEnodes = proxyPrivateKeys.map((x: string, i: number) => [
```